### PR TITLE
.nav implementation

### LINKED ISSUE #847: Visualize navmesh in map viewer
### Is your feature request related to a problem? Please describe.

https://developer.valvesoftware.com/wiki/Nav_Mesh
https://developer.valvesoftware.com/wiki/Half-Life:_Alyx_Workshop_Tools/Level_Design/Navigation_Mesh

It should be the .nav file next to the vmap.

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### PR DESCRIPTION
**Pull request by krotter0**
WIP, fixes #847 
Should work for all HLA and CS2 maps, need to test/implement for other games.
img

### CODE REVIEW

File: ValveResourceFormat/NavMesh/NavMeshFile.cs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+            if (Version >= 35)
+            {
+                ReadPostMetadata(binaryReader);
+
+                while (binaryReader.ReadByte() == 0)
+                {
+                    //Reading past 0x00 padding
+                }
+
+                var kv3 = new BinaryKV3();
```
Comment (kristiker) [inline ValveResourceFormat/NavMesh/NavMeshFile.cs:79]: I think we should store this, if there's any interesting data.

Comment (krotter0) [reply ValveResourceFormat/NavMesh/NavMeshFile.cs:79]: I'd definitely want to, just not sure how yet. I think it's game-specific data, CS2 stores hidingspotdata, spotencounterdata, approachdata, earliestoccupytime for each area.

**Comment by xPaw** [comment]
Add a small file to tests please

**Comment by xPaw** [comment]
Can you also add .nav support in decompiler, similar to .vcs files. Then we can scan them with --stats to test all files.

**Pull request merged**